### DISCUSSION TITLE
ESP32 Arduino 2.0.3 based on ESP-IDF 4.4.1

### DISCUSSION THREAD

**Discussion by me-no-dev** [espressif/arduino-esp32#6684] [Announcements]
## What's Changed
* Fixes String(float) issue with Stack Smashing by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6138
* Allows user to bypass PSRAM test and boot faster with WROVER by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6135
* Fixes UART MODBUS and Loopback issue by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6133
* Adds UART RX IRQ Callback with onReceive() by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6134
* NTP Examples: revert obsolete comment and updated Time example by @vortigont in https://github.com/espressif/arduino-esp32/pull/6073
* Add variantInit setups for adafruit boards by @ladyada in https://github.com/espressif/arduino-esp32/pull/6076
* Print.flush() - Arduino API conformance by @JAndrassy in https://github.com/espressif/arduino-esp32/pull/6084
* Add KSZ8041 support by @Darek7D in https://github.com/espressif/arduino-esp32/pull/6087
* BUGFIX - Sd check status by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6103
* Versions and feature Request Template updates by @VojtechBartoska in https://github.com/espressif/arduino-esp32/pull/6096
* Fix I2C Slave Compile by @mrengineer7777 in https://github.com/espressif/arduino-esp32/pull/6108
* Consistently change device index to singed integer in BluetoothSerial lib by @iltis42 in https://github.com/espressif/arduino-esp32/pull/6109
* WIFI STA AutoReconnect fix for assocfail reason,  Proposed fix for #2501 by @tablatronix in https://github.com/espressif/arduino-esp32/pull/6113
* Add Core Debug Level option to Tools menu for all boards in boards.txt by @gcweeks in https://github.com/espressif/arduino-esp32/pull/6110
* ci: Miscellaneous improvements by @Ouss4 in https://github.com/espressif/arduino-esp32/pull/6132
* IDF release/v4.4 by @me-no-dev in https://github.com/espressif/arduino-esp32/pull/6075
* Implement Ticker::active() by @everslick in https://github.com/espressif/arduino-esp32/pull/6148
* Add LionBit Dev Board. by @niwantha33 in https://github.com/espressif/arduino-esp32/pull/6151
* Add certificate bundle capability to WiFiClientSecure by @esp32wrangler in https://github.com/espressif/arduino-esp32/pull/6106
* Rename file pins_Arduino.h to pins_arduino.h by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6153
* Fix variant definition for w32-eth01 by @me-no-dev in https://github.com/espressif/arduino-esp32/pull/6159
* Fixing interrupts in LEDC by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6160
* Add Watchy board by @sqfmi in https://github.com/espressif/arduino-esp32/pull/6158
* .github/scripts: Fix indentation and trailing spaces. by @Ouss4 in https://github.com/espressif/arduino-esp32/pull/6157
* Fixes UART1 and UART2 default pins for ESP32-S3 by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6202
* Fix - SD mount issue by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6162
* [SSL] fix: restoring handshake timeout  by @vlastahajek in https://github.com/espressif/arduino-esp32/pull/6166
* Use 8.4.0+2021r2-patch2 toolchains for CI by @Jason2866 in https://github.com/espressif/arduino-esp32/pull/6184
* Fixes onReceive deadlock by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6201
* added explanatory comments to WebServer.h by @h-enes-simsek in https://github.com/espressif/arduino-esp32/pull/6204
* Add conditional include to WiFiProv.h by @mrengineer7777 in https://github.com/espressif/arduino-esp32/pull/6192
* Resolve WString TODO by @mrengineer7777 in https://github.com/espressif/arduino-esp32/pull/6190
* Allow HTTPCLIENT_1_1_COMPATIBLE to be disabled by @everslick in https://github.com/espressif/arduino-esp32/pull/6200
* Update RequestHandlersImpl.h to enable serveStatic on folders with LittleFS.  by @mathertel in https://github.com/espressif/arduino-esp32/pull/6179
* Refactor the CI scripts by @Ouss4 in https://github.com/espressif/arduino-esp32/pull/6191
* Fix random CaptivePortal.ino crashes by @im-pro-at in https://github.com/espressif/arduino-esp32/pull/6206
* Docs cleanup and version updated to 2.0.2 by @pedrominatel in https://github.com/espressif/arduino-esp32/pull/6213
* Unnecessary operation removed from map() in WMath.cpp by @IanSC in https://github.com/espressif/arduino-esp32/pull/6218
* Update of CameraWebServer example by @renebohne in https://github.com/espressif/arduino-esp32/pull/6219
* Touch Sensor IDF Refactoring by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6194
* Fix String.replace() broken by #6190 by @s-hadinger in https://github.com/espressif/arduino-esp32/pull/6224
* Add Github action for uploading arduino-esp32 component.  by @magicarm22 in https://github.com/espressif/arduino-esp32/pull/6203
* add feather esp32 v2 and qtpy c3 board def by @ladyada in https://github.com/espressif/arduino-esp32/pull/6223
* Added dual antenna for WiFi (based on the ESP32-WROOM-DA module) by @pedrominatel in https://github.com/espressif/arduino-esp32/pull/6226
* Remove pinMode in ledc for ESP32C3 by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6229
* Fixes Touch Sensor for ESP32-S3 and any future SoC by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6234
* HttpClient: Add cookie support (cookie jar) by @mattsches1 in https://github.com/espressif/arduino-esp32/pull/6216
* Support the updated MbedTLS in ESP-IDF v4.4 by @me-no-dev in https://github.com/espressif/arduino-esp32/pull/6243
* Fix compile with Arduino lib builder by @Jason2866 in https://github.com/espressif/arduino-esp32/pull/6244
* Fix CDC+JTAG is disabled when WiFi is used on ESP32-C3 by @me-no-dev in https://github.com/espressif/arduino-esp32/pull/6287
* Fix Check for _cookieJar in HTTPClient (#6266) by @bbsan2k in https://github.com/espressif/arduino-esp32/pull/6280
* Installing.rst_update by @VojtechBartoska in https://github.com/espressif/arduino-esp32/pull/6292
* Fixes softAPConfig() return by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6294
* Adds BLE examples to ESP32-C3 CI cycle by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6285
* Add Ethernet to CMakeLists by @copercini in https://github.com/espressif/arduino-esp32/pull/6261
* GPIO refactoring by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6259
* Added documentation for the Arduino IDE tools menu by @pedrominatel in https://github.com/espressif/arduino-esp32/pull/6284
* ADC esp32s2 attenuation fix for DAC pins by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6282
* Fixes UART pin setting + adds CTS/RTS HW Flow Control by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6272
* fixed http.end taking too long by @yigitabi5444 in https://github.com/espressif/arduino-esp32/pull/6277
* add lolin c3 mini board def by @wemos in https://github.com/espressif/arduino-esp32/pull/6306
* (boards.txt) Add partition scheme menu for AI Thinker ESP-CAM boards by @HACKER-3000 in https://github.com/espressif/arduino-esp32/pull/6310
* The M5Core2 and M5Tough have 40 pins by @pgrawehr in https://github.com/espressif/arduino-esp32/pull/6315
* add lolin s2 mini board def by @wemos in https://github.com/espressif/arduino-esp32/pull/6320
* Make USBHIDKeyboard::sendReport() public by @tobozo in https://github.com/espressif/arduino-esp32/pull/6322
* Update doc: Arduino as IDF component; fixes #5833 by @PilnyTomas in https://github.com/espressif/arduino-esp32/pull/6299
* Added ADC API doc + simple example by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6301
* [Docs] Added more details about embedded FLASH and PSRAM on ESP32-S3. by @pedrominatel in https://github.com/espressif/arduino-esp32/pull/6321
* LEDC & SigmaDelta documentation by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6330
* [Docs] Timer API documentation by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6335
* [Docs] DAC API documentation by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6337
* .github/workflows: Remove the selfhost runner test by @Ouss4 in https://github.com/espressif/arduino-esp32/pull/6343
* Add new board (Deneyap Kart 1A) by @zeynepdicle in https://github.com/espressif/arduino-esp32/pull/6324
* add lolin s2 pico board def by @wemos in https://github.com/espressif/arduino-esp32/pull/6325
* Update esp32-hal-log.h by @cyberman54 in https://github.com/espressif/arduino-esp32/pull/6351
* Fix some hardware definition pins by @ladyada in https://github.com/espressif/arduino-esp32/pull/6353
* Fix DNSServer crash by @BlueAndi in https://github.com/espressif/arduino-esp32/pull/6354
* Added more details about PlatformIO by @pedrominatel in https://github.com/espressif/arduino-esp32/pull/5540
* FIX - wifiBegin works every 2nd time when connecting to same AP by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6359
* WiFi DA: Added Dual Antenna to the docs and example created by @pedrominatel in https://github.com/espressif/arduino-esp32/pull/6357
* [esp32-hal-log.h] add #ifdef for LOG_LOCAL_LEVEL by @cyberman54 in https://github.com/espressif/arduino-esp32/pull/6358
* Added ESP32-WROOM-DA module to boards.txt by @pedrominatel in https://github.com/espressif/arduino-esp32/pull/6361
* FEATURE: Implemented new types of Wifi SmartConfig by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6367
* Fixes DHCP Server Lease Range for any AP Server Static IP Address by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6296
* Adds C++ std::function to Serial.onReceive() by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6364
* log_printf may cause memory leak and unexpected behaviour by @BlueAndi in https://github.com/espressif/arduino-esp32/pull/6360
* Build (using arduino-builder) and run tests by @Ouss4 in https://github.com/espressif/arduino-esp32/pull/6313
* Fixes rmtDeinit() and tests RX/TX before operations by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6369
* CameraWebServer duplicate line fix by @VojtechBartoska in https://github.com/espressif/arduino-esp32/pull/6370
* Fix ledc panic’ed when wrong setup of frequency and bit width by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6371
* Fixed the antenna GPIO argument on the DA example by @pedrominatel in https://github.com/espressif/arduino-esp32/pull/6392
* Allows setting only one pin (rx or tx) in the first begin() by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6394
* publish.yml: Remove the debug job and fix an error in the workflow. by @Ouss4 in https://github.com/espressif/arduino-esp32/pull/6408
* Implemented tone and noTone; fixes #980 by @PilnyTomas in https://github.com/espressif/arduino-esp32/pull/6402
* publish.yml:  Limit the running scope of the publish Workflow.  by @Ouss4 in https://github.com/espressif/arduino-esp32/pull/6428
* Add missing include in AsyncUDP.h by @mrengineer7777 in https://github.com/espressif/arduino-esp32/pull/6412
* Some board variant fixes by @ladyada in https://github.com/espressif/arduino-esp32/pull/6411
* Added another overloaded WiFiSTAClass::begin() function that provides… by @jpswensen in https://github.com/espressif/arduino-esp32/pull/6398
* [Docs] Added the guideline for documentation by @pedrominatel in https://github.com/espressif/arduino-esp32/pull/6409
* Fix boot freeze when trying to init PSRAM on Pico D4 by @s-hadinger in https://github.com/espressif/arduino-esp32/pull/6434
* Adds HardwareSerial::setTxBufferSize() by @gonzabrusco in https://github.com/espressif/arduino-esp32/pull/6383
* Fixes USB CDC setRxBufferSize(), begin(), _onRX() by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6413
* Change "python" to "python3" by @tomorrow56 in https://github.com/espressif/arduino-esp32/pull/6448
* Warns about SPP only available for ESP32 by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6455
* Edited VFSFileImpl::read to use both read/fread by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6456
* Adding sectorsize() and numSectors() to SD by @tobozo in https://github.com/espressif/arduino-esp32/pull/6457
* (boards.txt) Add partition scheme menu for WeMos WiFi&Bluetooth Battery by @garubi in https://github.com/espressif/arduino-esp32/pull/6479
* Add Preferences library API and tutorial documents by @Xylopyrographer in https://github.com/espressif/arduino-esp32/pull/6442
* Esp32 s3 support by @me-no-dev in https://github.com/espressif/arduino-esp32/pull/6341
* add variant init code for the feather s2 tft by @ladyada in https://github.com/espressif/arduino-esp32/pull/6447
* Adds HardwareSerial::setRxTimeout() by @gonzabrusco in https://github.com/espressif/arduino-esp32/pull/6397
* Fix linking failure for space in path in PlatformIO builder scripts by @maxgerhardt in https://github.com/espressif/arduino-esp32/pull/6464
* Add S3 in Headline by @Jason2866 in https://github.com/espressif/arduino-esp32/pull/6491
* Reordering - HardwareSerial Constructor by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6492
* IDF release/v4.4 by @me-no-dev in https://github.com/espressif/arduino-esp32/pull/6493
* Fix Reading MAC from efuse for ESP32 Arduino 2.x.x by @smuellener in https://github.com/espressif/arduino-esp32/pull/6459
* Update toolchain to gcc8_4_0-esp-2021r2-patch3 by @me-no-dev in https://github.com/espressif/arduino-esp32/pull/6497
* SDWebServer example fix for 2.x.x versions by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6505
* add package.json to release zip by @Jason2866 in https://github.com/espressif/arduino-esp32/pull/6501
* add S3 board in package_esp32_index.template.json by @Jason2866 in https://github.com/espressif/arduino-esp32/pull/6518
* Compiler error by @nimaltd in https://github.com/espressif/arduino-esp32/pull/6523
* BLEAddress: Correct less- and greater-than operators by @BrianRossmajer in https://github.com/espressif/arduino-esp32/pull/6521
* Fix Ticker::Active crash if _ticker null (#6511) by @GitNik1 in https://github.com/espressif/arduino-esp32/pull/6513
* Add better esptool support (ArduinoIDE 2.0.0) by @me-no-dev in https://github.com/espressif/arduino-esp32/pull/6506
* PHY8720(A) init timout failures: need to revert "GPIO refactoring (#6259)" by @dok-net in https://github.com/espressif/arduino-esp32/pull/6527
* IDF release/v4.4 by @me-no-dev in https://github.com/espressif/arduino-esp32/pull/6594
* Update camera example to support face detection and recognition by @me-no-dev in https://github.com/espressif/arduino-esp32/pull/6603
* Change pinMode OUTPUT to INPUT_OUTPUT by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6602
* Changed type of LEDC frequency from double to uint32_t by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6570
* BUGFIX: FS read + speed improvements for SD by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6569
* workflows/publish.yml: Run the workflow on success and failure only. by @Ouss4 in https://github.com/espressif/arduino-esp32/pull/6531
* Pull request/Issue Templates and Readme update by @VojtechBartoska in https://github.com/espressif/arduino-esp32/pull/6577
* PlatformIO minor updates by @valeros in https://github.com/espressif/arduino-esp32/pull/6625
* publish.yml: Minor fix to the workflow syntax by @Ouss4 in https://github.com/espressif/arduino-esp32/pull/6620
* LittleFs is working with C3 by @Jason2866 in https://github.com/espressif/arduino-esp32/pull/6618
* LittleFS Platformio example update by @Jason2866 in https://github.com/espressif/arduino-esp32/pull/6617
* Added RainMaker support on Arduino IDE for ESP32-C3/S2/S3 by @pedrominatel in https://github.com/espressif/arduino-esp32/pull/6598
* Touch change to init only selected GPIO. by @P-R-O-C-H-Y in https://github.com/espressif/arduino-esp32/pull/6609
* Allow BluetoothSerial::connect() with specified channel and more options by @ferbar in https://github.com/espressif/arduino-esp32/pull/6380
* Fixes stream load memory leak in WifiSecureClient for SSL CACert, Certificate, and Private Key by @fillybrese in https://github.com/espressif/arduino-esp32/pull/6387
* Call i2c_set_timeout in i2cSetClock by @PaulZC in https://github.com/espressif/arduino-esp32/pull/6537
* Uniform behaviour of WiFiClientSecure and WiFiClient setTimeout() by @gonzabrusco in https://github.com/espressif/arduino-esp32/pull/6562
* Salvage ESP_IDF TAGs when ArduinoCore logging is enabled by @ankostis in https://github.com/espressif/arduino-esp32/pull/6567
* add AirM2M_CORE_ESP32C3 board by @dreamcmi in https://github.com/espressif/arduino-esp32/pull/6613
* Added Unexpected Maker Reflow Master Pro (UM RMP) board + Fixed TinyS2 pins by @UnexpectedMaker in https://github.com/espressif/arduino-esp32/pull/6630
* Add CI job which tries to build arduino-esp32 as a component by @igrr in https://github.com/espressif/arduino-esp32/pull/5842
* [Docs] Actualize PlatformIO installation instructions by @valeros in https://github.com/espressif/arduino-esp32/pull/6629
* Fix HTTP Client failing to connect because of wrong timeout by @me-no-dev in https://github.com/espressif/arduino-esp32/pull/6633
* Added support for Trueverit ESP32 Universal IoT Driver MK III by @Simonefardella in https://github.com/espressif/arduino-esp32/pull/6640
* Fix onRequest being called for every slave address by @me-no-dev in https://github.com/espressif/arduino-esp32/pull/6649
* Add more controls over WiFi.begin and WiFi.scan by @me-no-dev in https://github.com/espressif/arduino-esp32/pull/6651
* Fixes INADDR_NONE  by @SuGlider in https://github.com/espressif/arduino-esp32/pull/6659
* Properly handle ARDUINO_PARTITION define in PlatformIO by @valeros in https://github.com/espressif/arduino-esp32/pull/6681
* Added new "Non TinyUF2 bootloader only" capability for boards that support TinyUF2 bootloaders by @UnexpectedMaker in https://github.com/espressif/arduino-esp32/pull/6668
* Fix uploader tool for ESPduino32 in boards.txt by @DeuxVis in https://github.com/espressif/arduino-esp32/pull/6665

## New Contributors
* @JAndrassy made their first contribution in https://github.com/espressif/arduino-esp32/pull/6084
* @Darek7D made their first contribution in https://github.com/espressif/arduino-esp32/pull/6087
* @mrengineer7777 made their first contribution in https://github.com/espressif/arduino-esp32/pull/6108
* @iltis42 made their first contribution in https://github.com/espressif/arduino-esp32/pull/6109
* @gcweeks made their first contribution in https://github.com/espressif/arduino-esp32/pull/6110
* @niwantha33 made their first contribution in https://github.com/espressif/arduino-esp32/pull/6151
* @esp32wrangler made their first contribution in https://github.com/espressif/arduino-esp32/pull/6106
* @sqfmi made their first contribution in https://github.com/espressif/arduino-esp32/pull/6158
* @h-enes-simsek made their first contribution in https://github.com/espressif/arduino-esp32/pull/6204
* @mathertel made their first contribution in https://github.com/espressif/arduino-esp32/pull/6179
* @im-pro-at made their first contribution in https://github.com/espressif/arduino-esp32/pull/6206
* @IanSC made their first contribution in https://github.com/espressif/arduino-esp32/pull/6218
* @renebohne made their first contribution in https://github.com/espressif/arduino-esp32/pull/6219
* @s-hadinger made their first contribution in https://github.com/espressif/arduino-esp32/pull/6224
* @magicarm22 made their first contribution in https://github.com/espressif/arduino-esp32/pull/6203
* @mattsches1 made their first contribution in https://github.com/espressif/arduino-esp32/pull/6216
* @bbsan2k made their first contribution in https://github.com/espressif/arduino-esp32/pull/6280
* @yigitabi5444 made their first contribution in https://github.com/espressif/arduino-esp32/pull/6277
* @pgrawehr made their first contribution in https://github.com/espressif/arduino-esp32/pull/6315
* @BlueAndi made their first contribution in https://github.com/espressif/arduino-esp32/pull/6354
* @jpswensen made their first contribution in https://github.com/espressif/arduino-esp32/pull/6398
* @gonzabrusco made their first contribution in https://github.com/espressif/arduino-esp32/pull/6383
* @tomorrow56 made their first contribution in https://github.com/espressif/arduino-esp32/pull/6448
* @garubi made their first contribution in https://github.com/espressif/arduino-esp32/pull/6479
* @Xylopyrographer made their first contribution in https://github.com/espressif/arduino-esp32/pull/6442
* @smuellener made their first contribution in https://github.com/espressif/arduino-esp32/pull/6459
* @nimaltd made their first contribution in https://github.com/espressif/arduino-esp32/pull/6523
* @BrianRossmajer made their first contribution in https://github.com/espressif/arduino-esp32/pull/6521
* @GitNik1 made their first contribution in https://github.com/espressif/arduino-esp32/pull/6513
* @ferbar made their first contribution in https://github.com/espressif/arduino-esp32/pull/6380
* @fillybrese made their first contribution in https://github.com/espressif/arduino-esp32/pull/6387
* @PaulZC made their first contribution in https://github.com/espressif/arduino-esp32/pull/6537
* @ankostis made their first contribution in https://github.com/espressif/arduino-esp32/pull/6567
* @dreamcmi made their first contribution in https://github.com/espressif/arduino-esp32/pull/6613
* @DeuxVis made their first contribution in https://github.com/espressif/arduino-esp32/pull/6665

**Full Changelog**: https://github.com/espressif/arduino-esp32/compare/2.0.2...2.0.3

<hr /><em>This discussion was created from the release <a href='https://github.com/espressif/arduino-esp32/releases/tag/2.0.3'>ESP32 Arduino 2.0.3 based on ESP-IDF 4.4.1</a>.</em>

**Comment by tablatronix** [comment]
How did you generate this ?!!!!

**Reply by VojtechBartoska** [reply]
@me-no-dev Maybe you click to announce this on Discussion as well? :) or some new feature of GitHub, I don't really know.

**Reply by me-no-dev** [reply]
yup! I clicked on "create discussion"

**Reply by kolinlin** [reply]
Why this version of app image not work with esp-idf 4.4.1 bootloader?

**Comment by kolinlin** [comment]
Why this version of app image not work with esp-idf 4.4.1 bootloader?

**Reply by VojtechBartoska** [reply]
@kolinlin I'm not sure what you mean. Can you provide more info? (consider opening a new issue)
